pkg/pod/status.go

```diff
@@ -290,8 +290,6 @@ func filterResultsAndResources(results []v1beta1.PipelineResourceResult, specRes
 		case v1beta1.InternalTektonResultType:
 			// Internal messages are ignored because they're not used as external result
 			continue
-		case v1beta1.PipelineResourceResultType:
```

Comment:
This function can be simplified now. The first return value is any TaskRunResults, the second value is any PipelineResourceResults of type PipelineResourceResultType or UnknownResultType, and the third return value is all PipelineResourceResults that aren't InternalTektonResultTypes.

I'm not sure UnknownResourceTypes are actually used anywhere. It might be simpler to just have one function that filters a list of PipelineResourceResults and returns any that are of type TaskRunResultType, and another function that turns those results into TaskRunResults.

Please also add a docstring.

Reply:
On second thought I think this can happen separately.